src/player.cpp

```diff
@@ -10790,6 +10795,7 @@ bool player::gunmod_remove( item &gun, item& mod )
 
     i_add_or_drop( mod );
     gun.contents.erase( iter );
+    gun.on_contents_changed();
```

Comment:
Gunmods aren't strictly speaking item contents. The gradual pattern is to move towards accessing them via `item::gunmods()`. As no code should be written to exploit this callback it shouldn't be added here as it's only encourages coupling between the systems when we should be aiming to separate them.